js/util/segment-fetcher.js

```diff
@@ -173,195 +87,33 @@ SegmentFetcher.DontVerifySegment = function(data)
  * exceptions.
  * @param {function} onError Call onError.onError(errorCode, message) for
  * timeout or an error processing segments. errorCode is a value from
- * SegmentFetcher.ErrorCode and message is a related string.
+ * PipelineFixed.ErrorCode and message is a related string.
  * NOTE: The library will log any exceptions thrown by this callback, but for
  * better error handling the callback should catch and properly handle any
  * exceptions.
+ *
+ * Example:
+ *     var onComplete = function(content) { ... }
+ *
+ *     var onError = function(errorCode, message) { ... }
+ *
+ *     var interest = new Interest(new Name("/data/prefix"));
+ *     interest.setInterestLifetimeMilliseconds(1000);
+ *     SegmentFetcher.fetch(face, interest, null, onComplete, onError);
  */
 SegmentFetcher.fetch = function
   (face, baseInterest, validatorKeyChainOrVerifySegment, onComplete, onError)
 {
+  var basePrefix = baseInterest.getName().toUri();
+
   if (validatorKeyChainOrVerifySegment == null ||
       validatorKeyChainOrVerifySegment instanceof KeyChain)
-    new SegmentFetcher
-      (face, validatorKeyChainOrVerifySegment, SegmentFetcher.DontVerifySegment,
+    new PipelineFixed
+      (basePrefix, face, validatorKeyChainOrVerifySegment,
        onComplete, onError)
       .fetchFirstSegment(baseInterest);
   else
-    new SegmentFetcher
-      (face, null, validatorKeyChainOrVerifySegment, onComplete, onError)
+    new PipelineFixed
```

Comment:
Given `verifySegment` is no longer supported, calling with that pattern should raise an error rather than result in skipped verification.

Reply:
No, if `validatorKeyChain` is not provided it means we do not want to validate the data. We do not force data validation, it should be optional. However, `validatorKeyChain` is provided and it has a problem then we need to raise an error (as the code does).

Reply:
Answer this question first: is `verifySegment` still supported?

YES => implement it in PipelineFixed

NO => check that validatorKeyChain is either null/undefined (indicating don’t verify) or a KeyChain instance; throw otherwise